src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -255,6 +257,66 @@ private static string ReplaceWith24HourClock(string format)
             return Regex.Replace(format, pattern, "HH${nonHours}", RegexOptions.IgnorePatternWhitespace);
         }
 
+        /// <summary>
+        /// Resolves month/minute ambiguity and 24h vs AM/PM.
+        /// </summary>
+        /// <param name="format">Format string.</param>
+        /// <returns></returns>
+        private static string ResolveDateTimeFormatAmbiguities(string format)
+        {
+            if (format.Length == 0)
+            {
+                return format;
+            }
+
+            // Handling especial cases
+            if (format.Length == 1)
+            {
+                switch (format)
+                {
+                    case "m":
+                    case "h":
+                        return $"%{format.ToUpperInvariant()}";
+                    default:
+                        return $"%{format}";
+                }
+            }
+
+            Regex minutesBeforeSecondsRegex = new Regex("[M]+[^dDyYhHmM]+[sS]", RegexOptions.Compiled);
+            Regex minutesAfterHoursRegex = new Regex("[hH][^dDyYmM]+[M]+", RegexOptions.Compiled);
+            Regex time12hRegex = new Regex("[aA][mM]\\/[pP][mM]|[aA]\\/[pP]", RegexOptions.Compiled);
+            Regex ampmReplaceRegex = new Regex("[aA][mM]\\/[pP][mM]", RegexOptions.Compiled);
+            Regex apReplaceRegex = new Regex("[aA]\\/[pP]", RegexOptions.Compiled);
+
+            var hourChar = time12hRegex.Match(format).Success ? 'h' : 'H';
+
+            format = ampmReplaceRegex.Replace(format, "tt");
+            format = apReplaceRegex.Replace(format, "t");
+            format = format.Replace('Y', 'y')
+                           .Replace('m', 'M') // All "m" char to upper case to express months
+                           .Replace('D', 'd')
+                           .Replace('H', hourChar)
+                           .Replace('h', hourChar)
+                           .Replace('S', 's')
+                           .Replace('F', 'f');
```

Comment:
>.Replace('F', 'f'); [](http://example.com/codeflow?start=27&length=19)

The sub-second specifier should be `0`, not `f` (compatible with Excel).

Reply:
This was intentionally left out since my next fix will involve the milliseconds formation.

Reply:
This has been fixed within my latest push.